File: Sem4/MPP/cs/lab8cs/Networking/dto/AngajatDTO.cs
using System.Text.Json.Serialization;

namespace Networking.dto;
using Domain;
public class AngajatDTO:EntityDTO
{
    [JsonInclude]
    public string Username{ get; set; }
    [JsonInclude]
    public string Password{ get; set; }
    [JsonInclude]
    public string Mail{ get; set; }

    [JsonConstructor]
    public AngajatDTO(string username, string password, string mail)
    {
        this.Username = username;
        this.Password = password;
        this.Mail = mail;
    }
    

    public static AngajatDTO fromAngajat(Angajat angajat)
    {
        var a=new AngajatDTO(angajat.Username,angajat.Password,angajat.Mail);
        a.Id = angajat.Id;
        return a;
    }

    public Angajat toAngajat()
    {
        var a=new Angajat(Username, Password, Mail);
        a.Id = Id;
        return a;
    }
    
}

File: Sem4/MPP/java/lab9client/Network/src/main/java/ubb/scs/mpp/Network/dto/AngajatDTO.java
package ubb.scs.mpp.Network.dto;

import ubb.scs.mpp.model.Angajat;

public class AngajatDTO extends EntityDTO {

    private final String username;
    private final String password;
    private final String mail;
    public AngajatDTO(final String username, final String password,final String mail) {
        this.username = username;
        this.password = password;
        this.mail = mail;
    }
    public String getUsername() {
        return username;
    }
    public String getPassword() {
        return password;
    }
    public String getMail() {
        return mail;
    }
    public static AngajatDTO fromAngajat(Angajat entity) {
            var a=new AngajatDTO(entity.getUsername(), entity.getPassword(), entity.getMail());
            a.setId(entity.getId());
            return a;
    }

    public Angajat toAngajat() {
        var a=new Angajat(getUsername(), getPassword(),getMail());
        a.setId(getId());
        return a;
    }
}
